src/lib/data/csvExport.ts

```diff
@@ -0,0 +1,39 @@
+export interface NormDataRecord {
+	ProveAar: string;
+	Opplegg: string;
+	Mikrobe: string;
+	Antibiotika: string;
+	region: string;
+	antall: number;
+	Andel_R: number;
+	Andel_S_I: number;
+	antall_R: number;
+	antall_S_I: number;
+}
+
+const DownloadCSV = (data: NormDataRecord[], fileName: string) => {
+	const convertToCSV = (data: NormDataRecord[]) => {
+		if (!data || data.length === 0) {
+			return "";
+		}
+
+		const header = Object.keys(data[0]).join(",");
+		const rows = data.map((row) =>
+			Object.values(row)
+				.map((value) => (typeof value === "string" ? `"${value}"` : value))
+				.join(",")
+		);
+		return [header, ...rows].join("\n");
+	};
+
+	const csvContent = convertToCSV(data);
+	const blob = new Blob([csvContent], { type: "text/csv;charset=utf-8;" });
+	const link = document.createElement("a");
+	link.href = URL.createObjectURL(blob);
+	link.download = `${fileName}.csv`;
+	document.body.appendChild(link);
+	link.click();
+	document.body.removeChild(link);
+};
+
+export default DownloadCSV;
```

Comment:
While this works in practice, better to make this into a react component. If you don't plan on reusing this right now, just embed it into the `MultiRegionResistanceLineCharts` component directly.

The standard practice for cleanup is to use `revokeObjectURL` instead of appending and removing the element

```js
const url = URL.createObjectURL(blob);
const link = document.createElement('a');
link.setAttribute('href', url);
link.setAttribute('download', `${fileName}.csv`);
link.click();

URL.revokeObjectURL(url);
```

Reply:
Thanks for suggestion Joshua, I will apply it. 

About the reusing this code. I will need this again, because Erik asked to download the data in E. Coli Dashboard too